File: java/leetcode/src/Task367.java
import org.junit.jupiter.api.Test;

public class Task367 {

    public boolean isPerfectSquare(int num) {
        int left = 1, right = num;
        int res=0;
        while (left <= right) {
            int mid = left + (right - left) / 2;
            long result = 1L* mid * mid;
            if (result == num) {
                return true;
            } else if (result > num) {
                right = mid - 1;

            } else {
                left = mid+1;
                res = mid;
            }

        }
        return false;
    }

    @Test
    void test(){
        System.out.println(isPerfectSquare(4));
    }
}

File: py/Task367.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        left, right = 1, num
        while left <= right:
            mid = (left + right) >> 1
            if mid * mid == num:
                return True
            elif mid * mid < num:
                left = mid + 1
            else:
                right = mid - 1
        return False


num = 14
obj = Solution()
print(obj.isPerfectSquare(num))
